.github/workflows/test.yaml

```diff
@@ -1,69 +1,82 @@
     strategy:
+      fail-fast: false
       matrix:
-        python-version: [ 2.7, 3.7 ]
-        ckan-version: [ 2.8, 2.9 ]
-        exclude:
-          - python-version: 3.7
-            ckan-version: 2.8
-    env:
-      CKAN_PATH: ./ckan
-      VIRTUAL_ENV: "1"  # let's fake a virtual environment, why not
+        python-version: [ 2.7, 3.9 ]
 
     steps:
       - uses: actions/checkout@v2
-      - name: Install Python
-        uses: actions/setup-python@v2
+      - uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python-version }}
-      - name: Set up the test environment
+      - name: Install requirements
+        run: pip install flake8 pycodestyle
+      - name: Check syntax
         run: |
-          make ckan-install CKAN_VERSION=${{ matrix.ckan-version }}
-          make create-test-db
-          make dev-setup
-      - name: Run tests
-        run: make test
+          flake8 . --count --max-line-length=127 --show-source --statistics
+
+  test:
+    name: CKAN
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - ckan-container-version: "2.9-py2"
+            ckan-postgres-version: "2.9"
+            ckan-solr-version: "2.9-solr8"
+            ckan-requirements: "requirements.py2.txt"
+          - ckan-container-version: "2.9"
+            ckan-postgres-version: "2.9"
+            ckan-solr-version: "2.9-solr8"
+            ckan-requirements: "requirements.py3.txt"
+          - ckan-container-version: "2.10"
+            ckan-postgres-version: "2.10"
+            ckan-solr-version: "2.10"
+            ckan-requirements: "requirements.py3.txt"
+
+    container:
+      image: openknowledge/ckan-dev:${{ matrix.ckan-container-version }}
+    services:
+      solr:
+        image: ckan/ckan-solr:${{ matrix.ckan-solr-version }}
+      postgres:
+        image: ckan/ckan-postgres-dev:${{ matrix.ckan-postgres-version }}
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: postgres
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+      redis:
+          image: redis:3
+    env:
+      CKAN_SQLALCHEMY_URL: postgresql://ckan_default:pass@postgres/ckan_test
+      CKAN_DATASTORE_WRITE_URL: postgresql://datastore_write:pass@postgres/datastore_test
+      CKAN_DATASTORE_READ_URL: postgresql://datastore_read:pass@postgres/datastore_test
+      CKAN_SOLR_URL: http://solr:8983/solr/ckan
+      CKAN_REDIS_URL: redis://redis:6379/1
+      CKAN_SITE_URL: http://localhost:5000
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install requirements
+      run: |
+        pip install -r ${{ matrix.ckan-requirements }}
+        pip install -r dev-${{ matrix.ckan-requirements }}
+        pip install git+https://github.com/fjelltopp/ckanext-emailasusername.git@development#egg=ckanext-emailasusername
+        pip install ckantoolkit
```

Comment:
ckantoolkit should be added to a requirements.txt file. Unless it's ckanext-emailasusername dependency then we should install the requirements.txt from the extension as well.

Reply:
Fixed